src/NzbDrone.Core/ImportLists/PTP/PTPRequestGenerator.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using Newtonsoft.Json;
+using NLog;
+using NzbDrone.Common.Http;
+
+namespace NzbDrone.Core.ImportLists.PTP;
+
+public class PTPRequestGenerator : IImportListRequestGenerator
+{
+    public PTPSettings Settings { get; set; }
+    public IHttpClient HttpClient { get; set; }
+    private IHttpRequestBuilderFactory RequestBuilder { get; set; }
+    public Logger Logger { get; set; }
+
+    public PTPRequestGenerator()
+    {
+    }
+
+    public virtual ImportListPageableRequestChain GetMovies()
+    {
+        var pageableRequests = new ImportListPageableRequestChain();
+
+        pageableRequests.Add(GetMoviesRequest());
+
+        return pageableRequests;
+    }
+
+    private IEnumerable<ImportListRequest> GetMoviesRequest()
+    {
+        RequestBuilder ??= new HttpRequestBuilder($"{Settings.BaseUrl.Trim().TrimEnd('/')}/collages.php")
+            .SetHeader("ApiUser", Settings.APIUser).SetHeader("ApiKey", Settings.APIKey)
+            .AddQueryParam("id", Settings.ID).AddQueryParam("action", "get_page").CreateFactory();
+
+        Logger.Info($"Importing PTP movies from collection: {Settings.ID}");
+
+        var requestBuilder = RequestBuilder.Create().AddQueryParam("page", 1);
+
+        var response = JsonConvert.DeserializeObject<CollectionPage>(HttpClient.Execute(requestBuilder.Build()).Content);
```

Comment:
The best way to avoid the extra call is to base this list on `ImportListBase` instead of `HttpImportListBase`. This eliminates the separate classes for request generation and parsing and allows you more freedom to do it how you need to. Otherwise we'll have to live with the extra call.